src/java/fr/paris/lutece/plugins/forms/web/admin/FormResponseJspBean.java

```diff
@@ -391,19 +403,109 @@ public String doFormResponseSummary( HttpServletRequest request )
     public synchronized String doSaveFormResponseSummary( HttpServletRequest request )
     {
         Form form = null;
-        
-        try 
+
+        try
         {
-			form = findFormFrom( request );
-		} 
+            form = findFormFrom( request );
+        }
         catch (FormNotFoundException e1)
         {
-			return getStepView( request );
-		}
-        
-    	return doSaveResponse(request, form);
+            return getStepView( request );
+        }
+
+        return doSaveResponse(request, form);
+    }
+
+    /**
+     * Adds an iteration
+     *
+     * @param request
+     *            the request
+     * @return the XPage
+     * @throws SiteMessageException
+     *             if there is an error during the iteration
+     * @throws UserNotSignedException
+     *             if the user is not signed in
+     */
+    @Action( value = ACTION_ADD_ITERATION )
+    public synchronized String doAddIteration( HttpServletRequest request ) throws SiteMessageException, UserNotSignedException
+    {
+        IsRequestComingFromAction = true;
+
+        try
+        {
+            boolean bSessionLost = isSessionLost( );
+            findFormFrom( request );
+            if ( bSessionLost )
```

Comment:
When the session expires, an exception is triggered at the Lutece-core level, and the user will be disconnected.

Reply:
it works thanks so I remove the useless tes